src/config.rs

```diff
@@ -1,358 +1,2028 @@
+            },
+            mask_hf: MaskingThresholds {
+                enr_transparent: 0.07,
+                enr_suppress: 0.1,
+                emr_transparent: 0.3,
+            },
+            max_inc_factor: 2.0,
+            max_dec_factor_lf: 0.25,
+        }
+    }
+}
+
+/// Configuration for dominant nearend speech detection
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "derive_serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "derive_serde", serde(default))]
+pub struct DominantNearendDetection {
+    /// Echo-to-noise ratio threshold
+    pub enr_threshold: f32,
+
+    /// Echo-to-noise ratio threshold for exiting detection state
+    pub enr_exit_threshold: f32,
+
+    /// Signal-to-noise ratio threshold
+    pub snr_threshold: f32,
+
+    /// Duration to hold detection state
+    pub hold_duration: i32,
+
+    /// Threshold for triggering detection
+    pub trigger_threshold: i32,
+
+    /// Whether to use during initial processing phase
+    pub use_during_initial_phase: bool,
+
+    /// Whether to use unbounded echo spectrum
+    pub use_unbounded_echo_spectrum: bool,
+}
+
+impl Default for DominantNearendDetection {
+    fn default() -> Self {
+        Self {
+            enr_threshold: 0.25,
+            enr_exit_threshold: 10.0,
+            snr_threshold: 30.0,
+            hold_duration: 50,
+            trigger_threshold: 12,
+            use_during_initial_phase: true,
+            use_unbounded_echo_spectrum: true,
+        }
+    }
+}
+
+/// Configuration for subband-based nearend detection
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "derive_serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "derive_serde", serde(default))]
+pub struct SubbandNearendDetection {
+    /// Number of blocks to average for nearend detection
+    pub nearend_average_blocks: usize,
+
+    /// First subband region configuration
+    pub subband1: SubbandRegion,
+
+    /// Second subband region configuration
+    pub subband2: SubbandRegion,
+
+    /// Nearend threshold
+    pub nearend_threshold: f32,
+
+    /// Signal-to-noise ratio threshold
+    pub snr_threshold: f32,
+}
+
+impl Default for SubbandNearendDetection {
+    fn default() -> Self {
+        Self {
+            nearend_average_blocks: 1,
+            subband1: SubbandRegion::default(),
+            subband2: SubbandRegion::default(),
+            nearend_threshold: 1.0,
+            snr_threshold: 1.0,
+        }
+    }
+}
+
+/// Configuration for a subband frequency region
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "derive_serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "derive_serde", serde(default))]
+pub struct SubbandRegion {
+    /// Lower frequency bound of the subband region
+    pub low: usize,
+
+    /// Upper frequency bound of the subband region
+    pub high: usize,
+}
+
+impl Default for SubbandRegion {
+    fn default() -> Self {
+        Self { low: 1, high: 1 }
```

Comment:
Is this what the defaults are in the C++ code?

Reply:
https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/blob/master/webrtc/api/audio/echo_canceller3_config.h#L216

Yes, I believe so